python-ecosys/requests/requests/__init__.py

```diff
@@ -90,7 +90,7 @@ def request(
     try:
         s.connect(ai[-1])
         if proto == "https:":
-            s = ussl.wrap_socket(s, server_hostname=host)
+            s = tls.SSLContext(tls.PROTOCOL_TLS_CLIENT).wrap_socket(s, server_hostname=host)
```

Comment:
Testing this change, it no longer works because `verify_mode` defaults to `CERT_REQUIRED`... I guess that's secure, but we really need to make this work like it did before.  So this needs to do `ctx.verify_mode = CERT_NONE` before wrapping the socket.

Similarly with the `umqtt` and `urllib` changes in this PR.

Reply:
I'll change the behavior for `requests` and `urllib`, as we don't ship a CA list by default we can't enable ssl verification without the user choosing it. (Previously, using the `ssl` library, with the legacy `ssl.wrap_socket` did disable verification).  For `umqtt`, I've adjusted the constructor to not take a boolean parameter for `ssl` anymore, but an sslcontext instead. So if someone wants to enable insecure ssl, they need to disable verification in the ssl context themselves (or load ca certificates or install a cert callback).

Ideally we should have a way to do that for `urllib` and `requests` as well. Looking at the documentation of their python counterpart, `urllib` has a `capath` parameter, and `requests` has a `verify` parameter, to point to a ca file. I'm not sure how strictly to `cpython` subset we adhere here, but I would like to implement parameters to allow passing a ca certificate for both modules.

Reply:
Oh, yes, `umqtt` now passes in the `SSLContext`, so no need to change the `verify_mode` for that.



> I'm not sure how strictly to `cpython` subset we adhere here, but I would like to implement parameters to allow passing a ca certificate for both modules.

Yes, that would be a good addition, allowing to pass in certs.  Ideally it would follow the CPython way.  Of course that's for a separate PR!